straxen/plugins/acqmon_processing.py

```diff
@@ -56,163 +137,307 @@ def compute(self, raw_records_aqmon):
-                            enumerate(['sum_wf', 'm_veto_sync',
-                                       'hev_stop', 'hev_start',
-                                       'he_stop', 'he_start',
-                                       'busy_stop', 'busy_start'])}
-        self.states = ['on', 'off']
+        return dtype
 
     def get_window_size(self):
-        return int(self.config['veto_proximity_window'] * 100)
-
-    def compute(self, events, aqmon_hits):
+        return self.veto_proximity_window
+
+    def set_result_for_veto(self,
+                            result_buffer: np.ndarray,
+                            event_window: np.ndarray,
+                            veto_intervals: np.ndarray,
+                            veto_name: str) -> None:
+        """
+        Fill the result buffer inplace. Goal is to find vetos with
+        <veto_name> that are either during, before or after the
+         current event_window.
+
+        :param result_buffer: The buffer to fill inplace
+        :param event_window: start/stop boundaries of the event to consider.
+            Should be an array with ['time'] and ['endtime'] which can be
+            based on event start/end times or S1/S2 times
+        :param veto_intervals: veto intervals datatype
+        :param veto_name: The name of the veto to fill the result buffer for
+        :return: Nothing, results are filled in place
+        """
+        # Set defaults to be some very long time
+        result_buffer[f'time_to_previous_{veto_name}'] = self.time_no_aqmon_veto_found
+        result_buffer[f'time_to_next_{veto_name}'] = self.time_no_aqmon_veto_found
+
+        selected_intervals = veto_intervals[veto_intervals['veto_type'] == f'{veto_name}_veto']
+        if not len(selected_intervals):
+            return
+
+        vetos_during_event = strax.touching_windows(selected_intervals,
+                                                    event_window)
+
+        # Figure out the vetos *during* an event
+        for event_i, veto_window in enumerate(vetos_during_event):
+            if veto_window[1] - veto_window[0]:
+                vetos_in_window = selected_intervals[veto_window[0]:
+                                                     veto_window[1]].copy()
+                starts = np.clip(vetos_in_window['time'],
+                                 event_window[event_i]['time'],
+                                 event_window[event_i]['endtime'])
+                stops = np.clip(vetos_in_window['endtime'],
+                                event_window[event_i]['time'],
+                                event_window[event_i]['endtime'])
+                # Now sum over all the stops-starts that are clipped
+                # within the duration of the event
+                result_buffer[event_i][f'veto_{veto_name}_overlap'] = np.sum(stops -
+                                                                             starts)
```

Comment:
So this refers to the fraction of time an event overlaps with a veto_interval is not it? Which should vary somewhere between 0 and left/right event extension per event?

Reply:
no it can be the entire event, why would it be limited to the extension of the vents?

Reply:
Ah sorry I should have been more precise. I meant if you assume no pile-up and that S2 is your triggering peak.

Reply:
ah yeah, this cannot be longer than your event and is usually 0